test/component_activated_menu_test.js

```diff
@@ -0,0 +1,322 @@
+
+    /* TODO: Test is on hold because we cannot use jsDom
+     * for testing mouse events
+     **/
+    it("should close the menu on mouseout (not sure if can test)");
+
+    it("should activate the config.selection_event on menu item selection", function() {
+      var value = 1;
+      $(document).on(TEST_SELECTION_EVENT_NAME, function() {
+        value++;
+      });
+
+      expect(value).to.equal(1);
+
+      menu.$node.find("li > a").eq(0).click();
+      expect(value).to.equal(2);
+    });
+  });
+
+  describe("ActivatedMenuContainer", function() {
+    it("should apply the config.container_id to $node", function() {
+      expect(menu.container.$node.attr("id")).to.equal(CONTAINER_ID);
+    });
+
+    it("should apply the config.container_classname to $node", function() {
+      var classes = CONTAINER_CLASSNAME.split(" ");
+      for(var i=0; i<classes.length; ++i) {
+        expect(menu.container.$node.hasClass(classes[i])).to.be.true;
+      }
+    });
+
+    it("should add the Container to the DOM", function() {
+      var $node = $("#" + CONTAINER_ID);
+      expect($node).to.exist;
+      expect($node.length).to.equal(1);
+    });
+
+    it("should wrap the original menu element", function() {
+      expect(menu.$node.parent().attr("id")).to.equal(CONTAINER_ID);
+    });
+
+    it("should make the menu public", function() {
+      expect(menu.container.menu).to.exist;
+      expect(menu.container.menu).to.equal(menu);
+    });
+
+    it("should make the $node public", function() {
+      expect(menu.container.$node).to.exist;
+      expect(menu.container.$node.length).to.equal(1);
+    });
+
+    it("should make the instance available as data on the $node", function() {
+      expect(menu.container.$node.data("instance")).to.exist;
+      expect(menu.container.$node.data("instance")).to.equal(menu.container);
+    });
+  });
+
+  describe("ActivatedMenuActivator (passed)", function() {
+    it("should use the passed activator when present in config", function() {
+      expect(menu.activator).to.exist;
+      expect(menu.activator.$node).to.exist;
+      expect(menu.activator.$node.length).to.equal(1);
+      expect(menu.activator.$node.get(0)).to.equal($("#" + ACTIVATOR_ID).get(0));
+    });
+
+    it("should make the $node public", function() {
+      expect(menu.activator.$node).to.exist;
+      expect(menu.activator.$node.length).to.equal(1);
+    });
+
+    it("should make the menu public", function() {
+      expect(menu.activator.menu).to.exist;
+      expect(menu.activator.menu).to.equal(menu);
+    });
+
+    it("should make the instance available as data on the $node", function() {
+      expect(menu.activator.$node.data("instance")).to.exist;
+      expect(menu.activator.$node.data("instance")).to.equal(menu.activator);
+    });
+
+    it("should open the menu when activator is clicked", function() {
+      // TODO... 
+      menu.activator.$node.click();
+      //expect(1).to.equal(0); // Fail until done.
```

Comment:
leaving these in to remind yourself for later?

Reply:
Nope. Looks like I got interrupted during writing the test.
I often start with something like:

```
it("should do this thing", function() {
  expect(1).to.equal(0); // Fail until done.
});
```

And then start turning all those horrible red fails into nice green passes. Looks like the `// TODO...` was probably me making a mark of where to pick up again, while commenting out the fail (usually just delete when done). Clearly this technique failed me for some reason. 

I shall revisit this test and complete it properly. Well spotted.